packages/starknet-snap/src/state/network-state-manager.test.ts

```diff
@@ -0,0 +1,243 @@
+
+      const stateManager = new NetworkStateManager();
+      const result = await stateManager.findNetwork({
+        chainId,
+      });
+
+      expect(result).toStrictEqual(STARKNET_SEPOLIA_TESTNET_NETWORK);
+    });
+
+    it('returns null if the token can not be found', async () => {
+      const chainId = constants.StarknetChainId.SN_SEPOLIA;
+      await mockState({
+        networks: [STARKNET_MAINNET_NETWORK],
+      });
+
+      const stateManager = new NetworkStateManager();
+      const result = await stateManager.findNetwork({
+        chainId,
+      });
+
+      expect(result).toBeNull();
+    });
+  });
+
+  describe('addDefaultNetworks', () => {
+    it("adds the networks if the given networks don't exist", async () => {
+      const { state } = await mockState({
+        networks: [],
+      });
+
+      const stateManager = new NetworkStateManager();
+      await stateManager.addDefaultNetworks([
+        STARKNET_MAINNET_NETWORK,
+        STARKNET_SEPOLIA_TESTNET_NETWORK,
+      ]);
+
+      expect(state.networks).toStrictEqual([
+        STARKNET_MAINNET_NETWORK,
+        STARKNET_SEPOLIA_TESTNET_NETWORK,
+      ]);
+    });
+
+    it('updates the networks if the given networks found', async () => {
+      const { state } = await mockState({
+        networks: [STARKNET_MAINNET_NETWORK],
+      });
+
+      const stateManager = new NetworkStateManager();
+      await stateManager.addDefaultNetworks([
+        {
+          ...STARKNET_MAINNET_NETWORK,
+          name: 'Updated Network',
+        },
+        STARKNET_SEPOLIA_TESTNET_NETWORK,
+      ]);
+
+      expect(state.networks[0]).toStrictEqual({
+        ...STARKNET_MAINNET_NETWORK,
+        name: 'Updated Network',
+      });
+      expect(state.networks[1]).toStrictEqual(STARKNET_SEPOLIA_TESTNET_NETWORK);
+    });
+
+    it('throws a `StateManagerError` if an error was thrown', async () => {
+      const { getDataSpy } = await mockState({
+        networks: [STARKNET_MAINNET_NETWORK],
+      });
+      getDataSpy.mockRejectedValue(new Error('Error'));
+
+      const stateManager = new NetworkStateManager();
+
+      await expect(
+        stateManager.addDefaultNetworks([
+          {
+            ...STARKNET_MAINNET_NETWORK,
+            name: 'Updated Network',
+          },
+          STARKNET_SEPOLIA_TESTNET_NETWORK,
+        ]),
+      ).rejects.toThrow(StateManagerError);
+    });
+  });
+
+  describe('list', () => {
+    it('returns the list of network by chainId', async () => {
+      const chainId = constants.StarknetChainId.SN_SEPOLIA;
+      await mockState({
+        networks: [STARKNET_MAINNET_NETWORK, STARKNET_SEPOLIA_TESTNET_NETWORK],
+      });
+
+      const stateManager = new NetworkStateManager();
+      const result = await stateManager.list([new ChainIdFilter([chainId])]);
+
+      expect(result).toStrictEqual([STARKNET_SEPOLIA_TESTNET_NETWORK]);
+    });
+
+    it('returns empty array if the account address can not be found', async () => {
```

Comment:
account address => network chainId. 

`it('returns empty array if a network with the given chainId cannot be found')`